src/PIL/ImageMode.py

```diff
@@ -36,43 +39,52 @@ def getmode(mode):
     if not _modes:
         # initialize mode cache
         modes = {}
-        for m, (basemode, basetype, bands) in {
+        for m, (basemode, basetype, bands, typestr) in {
             # core modes
-            "1": ("L", "L", ("1",)),
-            "L": ("L", "L", ("L",)),
-            "I": ("L", "I", ("I",)),
-            "F": ("L", "F", ("F",)),
-            "P": ("P", "L", ("P",)),
-            "RGB": ("RGB", "L", ("R", "G", "B")),
-            "RGBX": ("RGB", "L", ("R", "G", "B", "X")),
-            "RGBA": ("RGB", "L", ("R", "G", "B", "A")),
-            "CMYK": ("RGB", "L", ("C", "M", "Y", "K")),
-            "YCbCr": ("RGB", "L", ("Y", "Cb", "Cr")),
-            "LAB": ("RGB", "L", ("L", "A", "B")),
-            "HSV": ("RGB", "L", ("H", "S", "V")),
+            # Bits need to be extended to bytes
+            "1": ("L", "L", ("1",), "|b1"),
+            "L": ("L", "L", ("L",), "|u1"),
+            "I": ("L", "I", ("I",), Image._ENDIAN + "i4"),
+            "F": ("L", "F", ("F",), Image._ENDIAN + "f4"),
+            "P": ("P", "L", ("P",), "|u1"),
+            "RGB": ("RGB", "L", ("R", "G", "B"), "|u1"),
+            "RGBX": ("RGB", "L", ("R", "G", "B", "X"), "|u1"),
+            "RGBA": ("RGB", "L", ("R", "G", "B", "A"), "|u1"),
+            "CMYK": ("RGB", "L", ("C", "M", "Y", "K"), "|u1"),
+            "YCbCr": ("RGB", "L", ("Y", "Cb", "Cr"), "|u1"),
+            # UNDONE - unsigned |u1i1i1
+            "LAB": ("RGB", "L", ("L", "A", "B"), "|u1"),
+            "HSV": ("RGB", "L", ("H", "S", "V"), "|u1"),
             # extra experimental modes
-            "RGBa": ("RGB", "L", ("R", "G", "B", "a")),
-            "BGR;15": ("RGB", "L", ("B", "G", "R")),
-            "BGR;16": ("RGB", "L", ("B", "G", "R")),
-            "BGR;24": ("RGB", "L", ("B", "G", "R")),
-            "BGR;32": ("RGB", "L", ("B", "G", "R")),
-            "LA": ("L", "L", ("L", "A")),
-            "La": ("L", "L", ("L", "a")),
-            "PA": ("RGB", "L", ("P", "A")),
+            "RGBa": ("RGB", "L", ("R", "G", "B", "a"), None),
+            "BGR;15": ("RGB", "L", ("B", "G", "R"), None),
+            "BGR;16": ("RGB", "L", ("B", "G", "R"), None),
+            "BGR;24": ("RGB", "L", ("B", "G", "R"), None),
+            "BGR;32": ("RGB", "L", ("B", "G", "R"), None),
```

Comment:
```suggestion
            "BGR;32": ("RGB", "L", ("B", "G", "R"), "u4"),
```

Not sure if this is big-endian or little-endian (nor if pillow would allow conversion from this mode to array), but I guess it should be a 4-byte uint.

Reply:
Ok, I've updated the entry for this.